### ISSUE TITLE
[BUG] Sample policy blocking pods created from CronJob / Job

### ISSUE THREAD

**Issue by rubroboletus** [kyverno/policies#156]
**Software version numbers**
1.5.1

* Kubernetes version: 1.21
* Kubernetes platform (if applicable; ex., EKS, GKE, OpenShift): EKS
* Kyverno version: 1.5.1

**Describe the bug**
When I apply sample policy - [https://kyverno.io/policies/best-practices/require_probes/require_probes/](https://kyverno.io/policies/best-practices/require_probes/require_probes/) - is blocking pods created from CronJob / Job.

**To Reproduce**
1. apply mentioned clusterpolicy
2. create any CronJob / Job using kubectl
3. pod from this CronJob / Job will be blocked by this sample clusterpolicy, as CronJobs and Jobs cannot have pod probes

**Expected behavior**
Policy should not check Pods, just Deployments, Daemonsets and StatefulSets, or there must be possible to skip this check for pods, created from CronJob / Job.
Is is duplicate of closed bugreport https://github.com/kyverno/policies/issues/113, I hope, now is all explained better.

**Contributor Resources**
- [Contributing Guidelines](https://github.com/kyverno/kyverno/blob/main/CONTRIBUTING.md)
